source/dub/internal/vibecompat/inet/url.d

```diff
@@ -66,45 +66,40 @@ struct URL {
 			str = str[idx+1 .. $];
 			bool requires_host = false;
 
-			switch(m_schema){
-				case "http":
-				case "https":
-				case "ftp":
-				case "spdy":
-				case "sftp":
-				case "file":
-					// proto://server/path style
-					enforce(str.startsWith("//"), "URL must start with proto://...");
-					requires_host = true;
-					str = str[2 .. $];
-					goto default;
-				default:
-					auto si = str.countUntil('/');
-					if( si < 0 ) si = str.length;
-					auto ai = str[0 .. si].countUntil('@');
-					sizediff_t hs = 0;
-					if( ai >= 0 ){
-						hs = ai+1;
-						auto ci = str[0 .. ai].countUntil(':');
-						if( ci >= 0 ){
-							m_username = str[0 .. ci];
-							m_password = str[ci+1 .. ai];
-						} else m_username = str[0 .. ai];
-						enforce(m_username.length > 0, "Empty user name in URL.");
-					}
-
-					m_host = str[hs .. si];
-					auto pi = m_host.countUntil(':');
-					if(pi > 0) {
-						enforce(pi < m_host.length-1, "Empty port in URL.");
-						m_port = to!ushort(m_host[pi+1..$]);
-						m_host = m_host[0 .. pi];
-					}
-
-					enforce(!requires_host || m_schema == "file" || m_host.length > 0,
-							"Empty server name in URL.");
-					str = str[si .. $];
+			auto schema_parts = m_schema.split("+");
+			if (schema_parts.any!(p => p.canFind("http", "https", "ftp", "spdy", "file", "sftp")))
```

Comment:
Shouldn't this be just `.back`?

Reply:
Also as the list of valid protocol is repeated, maybe a good idea to define them as an AliasSeq globally?

Reply:
I do not really know AliasSeq but what is the benefit of AliasSeq compared to a static const string array? As far as I see AliasSeq allows a list of values of different types but that seems not be necessary here.

Reply:
It auto-expands:

```diff
diff --git a/source/dub/internal/vibecompat/inet/url.d b/source/dub/internal/vibecompat/inet/url.d
index 54c3699..a40f691 100644
--- a/source/dub/internal/vibecompat/inet/url.d
+++ b/source/dub/internal/vibecompat/inet/url.d
@@ -67,7 +67,9 @@ struct URL {
 			bool requires_host = false;
 
 			auto schema_parts = m_schema.split("+");
-			if (schema_parts.any!(p => p.canFind("http", "https", "ftp", "spdy", "file", "sftp")))
+			import std.meta : AliasSeq;
+			alias schemes = AliasSeq!("http", "https", "ftp", "spdy", "file", "sftp");
+			if (schema_parts.any!(p => p.canFind(schemes)))
 			{
 				// proto://server/path style
 				enforce(str.startsWith("//"), "URL must start with proto://...");
```